src/mcts/node.cc

```diff
@@ -342,11 +348,14 @@ void NodeTree::ResetToPosition(const std::string& starting_fen,
     if (old_head == current_head_) seen_old_head = true;
   }
 
-  // If we didn't see old head, it means that new position is shorter.
+  // If we didn't see old head, it means that new position is shorter; or if the
+  // node is already marked as terminal, clear it to allow forced analysis.
   // As we killed the search tree already, trim it to redo the search.
```

Comment:
This part "As we killed the search tree" is related to the first part of previous sentence, now it's not clear.